src/ImmutablePropTypes.js

```diff
@@ -50,7 +50,7 @@ function getPropType(propValue) {
 function createChainableTypeChecker(validate) {
   function checkType(isRequired, props, propName, componentName, location) {
     componentName = componentName || ANONYMOUS;
-    if (props[propName] == null) {
+    if (props[propName] === null) {
```

Comment:
This needs to be `==` not `===` because it should check against `undefined` too.